src/resources/js/dependency.js

```diff
@@ -72,6 +75,7 @@
 						}
 					} else {
 						$dependent
+							.hide()
```

Comment:
So could you check if adding the class hides it or not `.is( ':hidden' )`? And based on that conditional use `.hide()/.show()`?